seungjong/utils/authorization.js

```diff
@@ -0,0 +1,16 @@
+const jwt = require("jsonwebtoken");
+
+const authorization = async (req, res, next) => {
+  try {
+    const token = req.headers.authorization;
+
+    if (!token) return res.status(400).json({ message: "TOKEN_EMPTY" });
+
+    const decoded = jwt.verify(token, process.env.SECRETKEY);
+    req.userId = decoded.userId;
```

Comment:
```suggestion
    const decoded = jwt.verify(token, process.env.SECRETKEY);
```
```suggestion
    const decoded = await jwt.verify(token, process.env.JWT_SECRETKEY);
    
    const user = await userService.getUserById(decoded.id)

    if (!user) {
	const error = new Error('USER_DOES_NOT_EXIST')
	error.statusCode = 404
		
	return res.status(error.statusCode).json({message: error.message})
    }
    
    req.userId = user.id;
```
토큰 검증 후에는 해당 토큰의 주인이 여전히 서버에서 유효한 회원인지를 검증하는 과정 또한 필요합니다. 인가처리는 보안상 민감한 부분이므로 로직이 꼼꼼히 작성될수록 좋습니다-!